lib/screens/profile/contact_support_public.dart

```diff
@@ -0,0 +1,292 @@
+
+import '../../l10n/app_localizations.dart';
+
+class ContactSupportPublicScreen extends HookConsumerWidget {
+  const ContactSupportPublicScreen({super.key});
+
+  @override
+  Widget build(BuildContext context, WidgetRef ref) {
+    final nameController = useTextEditingController();
+    final emailController = useTextEditingController();
+    final messageController = useTextEditingController();
+    final includeDeviceInfo = useRef(true);
+    final isOnInputPage = useState(true);
+    final formKey = useMemoized(() => GlobalKey<FormState>());
+    final sendSupportEmailFuture = useState<Future<QueryResult>?>(null);
+    final sendSupportEmailSnapshot = useFuture(sendSupportEmailFuture.value);
+    final deviceInfo = ref.watch(deviceInfoProvider).valueOrNull;
+    useListenableSelector(nameController, () => nameController.text.isEmpty);
+    useListenableSelector(emailController, () => emailController.text.isEmpty);
+    useListenableSelector(messageController, () => messageController.text.isEmpty);
+
+    void onIncludeDeviceInfoChange(bool state) => includeDeviceInfo.value = state;
+
+    void onTryAgain() => isOnInputPage.value = true;
+
+    String getDeviceInfoHtml() {
+      final rows = deviceInfo?.entries.map((entry) => '<th>${entry.key}</th><td>${entry.value}</td>').join('\n');
+      return '<table><tbody>$rows</tbody></table>';
+    }
+
+    Future<QueryResult> getSendSupportEmailFuture() {
+      final appVer = deviceInfo?.appVer;
+      final os = deviceInfo?.os;
+      return ref.read(gqlClientProvider).mutate$sendSupportEmail(
+            Options$Mutation$sendSupportEmail(
+              variables: Variables$Mutation$sendSupportEmail(
+                title: 'BTV $appVer $os',
+                content: '',
+                html: '<p>${messageController.text}</p>${includeDeviceInfo.value ? getDeviceInfoHtml() : ''}',
+                options: Input$EmailOptions(name: nameController.text, email: emailController.text),
+              ),
+            ),
+          );
+    }
+
+    void onSend() {
+      FocusManager.instance.primaryFocus?.unfocus();
+
+      if (formKey.currentState?.validate() != true) {
+        return;
+      }
+
+      sendSupportEmailFuture.value = getSendSupportEmailFuture();
+      isOnInputPage.value = false;
+    }
+
+    final Widget body;
+    if (isOnInputPage.value) {
+      body = Form(
+        key: formKey,
+        child: _InputPage(
+          nameController: nameController,
+          emailController: emailController,
+          messageController: messageController,
+          onIncludeDeviceInfoChange: onIncludeDeviceInfoChange,
+        ),
+      );
+    } else if (sendSupportEmailSnapshot.hasError || sendSupportEmailSnapshot.data?.hasException == true) {
+      body = ContactSupportFailure(onTryAgain: onTryAgain);
+    } else if (sendSupportEmailSnapshot.hasData) {
+      body = const ContactSupportSuccess();
+    } else {
+      body = const ContactSupportSendingIndicator();
+    }
```

Comment:
the response doesn't need to have data, and errors usually persist in the snapshot across "try again", so you won't see "loading.." when clicking try again.

So I usually prefer this ordering of the logic:
if (hasError) && !waiting
else if waiting
else success